src/modules/graphics/wrap_GraphicsShader.lua

```diff
@@ -324,9 +324,16 @@ local function createShaderStageCode(stage, code, lang, gles, glsl1on3, gammacor
 		GLSL.FUNCTIONS,
 		GLSL[stage].FUNCTIONS,
 		custom and GLSL[stage].MAIN_CUSTOM or GLSL[stage].MAIN,
-		((lang == "glsl1" or glsl1on3) and not gles) and "#line 0" or "#line 1",
-		code,
 	}
+	if defines then
+		for k, v in pairs(defines) do
```

Comment:
Iterating over the whole table that's passed in and treating every single field as a #define prevents the table from having other uses. I'm not sure if the table *should* have other uses, or if there's anything else it could be used for, but it's worth thinking about before locking ourselves out of that possibility.